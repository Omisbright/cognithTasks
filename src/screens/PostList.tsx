// import React from 'react';
// import {View, FlatList, Text} from 'react-native';
// import axios from 'axios';
// import Post, {PostProps} from '../components/Post';
// import Pagination from '../components/Pagination';
// import SearchBar from '../components/SearchBar';
// import {postListStyles} from './styles';
// import {AppProps} from '../navigation/types';

// type State = {
//   posts: PostProps[];
//   displayedPosts: PostProps[];
//   currentPage: number;
//   totalPages: number;
//   searchText: string;
//   interval: number | undefined;
//   error: string | undefined;
// };

// const SEARCH_PAGE = 0;

// export class PostList extends React.Component<AppProps<'PostList'>, State> {
//   constructor(props: AppProps<'PostList'>) {
//     super(props);
//     this.state = {
//       posts: [],
//       displayedPosts: [],
//       currentPage: SEARCH_PAGE,
//       totalPages: 0,
//       searchText: '',
//       interval: undefined,
//       error: undefined,
//     };
//   }

//   componentDidMount() {
//     this.fetchPosts(this.state.currentPage);
//     const interval = setInterval(() => {
//       this.fetchPosts(this.state.currentPage);
//     }, 10000);
//     this.setState({interval});
//   }

//   componentWillUnmount() {
//     if (this.state.interval) {
//       clearInterval(this.state.interval);
//     }
//   }

//   fetchPosts = async (page: number) => {
//     try {
//       const response = await axios.get(
//         `https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${page}`,
//       );
//       const {hits, nbPages} = response.data;
//       const uniqueArray = hits.filter(
//         (item: {objectID: string}, index: number, self: PostProps[]) =>
//           index ===
//           self.findIndex((t: {objectID: any}) => t.objectID === item.objectID),
//       );

//       this.setState({
//         posts: uniqueArray,
//         totalPages: nbPages,
//         currentPage: this.state.currentPage + 1,
//       });
//       this.setState({displayedPosts: uniqueArray});
//     } catch (error) {
//       console.log('Error fetching posts:', error);
//       this.setState({error: 'Failed to fetch posts'});
//     }
//   };

//   renderItem = ({item}: {item: PostProps}) => {
//     return (
//       <Post
//         title={item.title}
//         url={item.url}
//         created_at={item.created_at}
//         _tags={item._tags}
//         author={item.author}
//         objectID={item.objectID}
//         item={item}
//         navigation={this.props.navigation}
//       />
//     );
//   };

//   emptyComponent = () => {
//     return (
//       <View style={postListStyles.emptyComponentContainer}>
//         <Text style={postListStyles.emptyComponentText}>List is Empty</Text>
//       </View>
//     );
//   };

//   renderError = () => {
//     const {error} = this.state;
//     if (error) {
//       return (
//         <View style={postListStyles.errorContainer}>
//           <Text testID="error-message" style={postListStyles.errorText}>
//             {error}
//           </Text>
//         </View>
//       );
//     }
//     return null;
//   };

//   handlePageChange = (page: number) => {
//     this.setState({currentPage: page}, () => {
//       this.fetchPosts(page);
//     });
//   };

//   onChangeSearchText = (text: string) => {
//     this.setState({searchText: text});
//     const {posts} = this.state;
//     const filteredArray = posts.filter(
//       (element: {author: string; title: string}) => {
//         let searchableText = `${element.author} ${element.title}`.toLowerCase();
//         let searchText = text.toLowerCase();
//         return searchableText.includes(searchText);
//       },
//     );
//     this.setState({displayedPosts: filteredArray});
//   };

//   render() {
//     const {searchText, currentPage, totalPages, displayedPosts} = this.state;
//     return (
//       <View style={postListStyles.container}>
//         {this.renderError()}
//         <SearchBar
//           searchText={searchText}
//           onChangeSearchText={this.onChangeSearchText}
//         />
//         <FlatList
//           data={displayedPosts}
//           renderItem={this.renderItem}
//           keyExtractor={item => item.objectID}
//           ListEmptyComponent={this.emptyComponent}
//         />
//         <Pagination
//           currentPage={currentPage}
//           totalPages={totalPages}
//           onChangePage={this.handlePageChange}
//         />
//       </View>
//     );
//   }
// }

// export default PostList;

import React from 'react';
import {View, FlatList, Text} from 'react-native';
import axios from 'axios';
import Post, {PostProps} from '../components/Post';
import Pagination from '../components/Pagination';
import SearchBar from '../components/SearchBar';
import {postListStyles} from './styles';
import {AppProps} from '../navigation/types';

type State = {
  posts: PostProps[];
  displayedPosts: PostProps[];
  currentPage: number;
  totalPages: number;
  searchText: string;
  interval: number | undefined;
  error: string | undefined;
  loading: boolean;
};

const SEARCH_PAGE = 0;

export class PostList extends React.Component<AppProps<'PostList'>, State> {
  constructor(props: AppProps<'PostList'>) {
    super(props);
    this.state = {
      posts: [],
      displayedPosts: [],
      currentPage: SEARCH_PAGE,
      totalPages: 0,
      searchText: '',
      interval: undefined,
      error: undefined,
      loading: false,
    };
  }

  componentDidMount() {
    this.fetchPosts(this.state.currentPage);
    const interval = setInterval(() => {
      this.fetchPosts(this.state.currentPage + 1);
    }, 10000);
    this.setState({interval});
  }

  componentWillUnmount() {
    if (this.state.interval) {
      clearInterval(this.state.interval);
    }
  }

  fetchPosts = async (page: number) => {
    this.setState({loading: true});
    try {
      const response = await axios.get(
        `https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${page}`,
      );
      this.setState({loading: false});
      const {hits, nbPages} = response.data;
      const uniqueArray = hits.filter(
        (item: {objectID: string}, index: number, self: PostProps[]) =>
          index ===
          self.findIndex((t: {objectID: any}) => t.objectID === item.objectID),
      );

      this.setState({
        posts: uniqueArray,
        totalPages: nbPages,
        currentPage: page,
      });
      this.setState({displayedPosts: uniqueArray});
    } catch (error) {
      this.setState({loading: false});
      console.log('Error fetching posts:', error);
      this.setState({error: 'Failed to fetch posts'});
    }
  };

  renderItem = ({item, index}: {item: PostProps; index: number}) => {
    return (
      <Post
        title={item.title}
        url={item.url}
        created_at={item.created_at}
        _tags={item._tags}
        author={item.author}
        objectID={item.objectID}
        item={item}
        navigation={this.props.navigation}
        testID={`postItem.${index}`}
      />
    );
  };

  emptyComponent = () => {
    return (
      <View style={postListStyles.emptyComponentContainer}>
        <Text style={postListStyles.emptyComponentText}>List is Empty</Text>
      </View>
    );
  };

  renderError = () => {
    const {error} = this.state;
    if (error) {
      return (
        <View style={postListStyles.errorContainer}>
          <Text testID="error-message" style={postListStyles.errorText}>
            {error}
          </Text>
        </View>
      );
    }
    return null;
  };

  handlePageChange = (page: number) => {
    this.setState({currentPage: page}, () => {
      this.fetchPosts(page);
    });
  };

  onChangeSearchText = (text: string) => {
    this.setState({searchText: text});
    const {posts} = this.state;
    const filteredArray = posts.filter(
      (element: {author: string; title: string}) => {
        let searchableText = `${element.author} ${element.title}`.toLowerCase();
        let searchText = text.toLowerCase();
        return searchableText.includes(searchText);
      },
    );
    this.setState({displayedPosts: filteredArray});
  };

  render() {
    const {searchText, currentPage, totalPages, displayedPosts} = this.state;
    return (
      <View style={postListStyles.container}>
        <SearchBar
          searchText={searchText}
          onChangeSearchText={this.onChangeSearchText}
        />
        <FlatList
          data={displayedPosts}
          renderItem={this.renderItem}
          keyExtractor={item => item.objectID}
          ListEmptyComponent={!this.state.error ? this.emptyComponent : null}
          onEndReached={() => {
            if (this.state.loading) {
              return;
            }
            this.fetchPosts(this.state.currentPage + 1);
          }}
          onEndReachedThreshold={0.5}
        />
        {this.renderError()}
        <Pagination
          currentPage={currentPage}
          totalPages={totalPages}
          onChangePage={this.handlePageChange}
        />
      </View>
    );
  }
}

export default PostList;
