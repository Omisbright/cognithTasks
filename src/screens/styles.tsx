import {StyleSheet} from 'react-native';

export const postListStyles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingHorizontal: 16,
  },
  emptyComponentContainer: {
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  emptyComponentText: {
    color: 'red',
  },
  errorContainer: {
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  errorText: {
    color: 'red',
  },
  flatListContainer: {
    flex: 1,
  },
});

export const displayPageStyles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 16,
    backgroundColor: '#fff',
  },
  text: {
    marginVertical: 3,
  },
});
