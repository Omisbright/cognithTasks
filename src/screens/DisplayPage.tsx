import {Text, View} from 'react-native';
import React, {Component} from 'react';
import {AppProps} from '../navigation/types';
import {displayPageStyles} from './styles';

export default class DisplayPage extends Component<AppProps<'DisplayPage'>> {
  constructor(props: AppProps<'DisplayPage'>) {
    super(props);
  }
  render() {
    const item = this.props?.route?.params?.item;
    return (
      <View style={displayPageStyles.container}>
        <Text style={displayPageStyles.text}>Title: {item.title}</Text>
        <Text style={displayPageStyles.text}>URL: {item.url}</Text>
        <Text style={displayPageStyles.text}>Author: {item.author}</Text>
        <Text style={displayPageStyles.text}>
          Created At: {item.created_at}
        </Text>
        <Text style={displayPageStyles.text}>
          Created At (i): {item.created_at_i}
        </Text>
        <Text style={displayPageStyles.text}>
          Number of Comments: {item.num_comments}
        </Text>
        <Text style={displayPageStyles.text}>Object ID: {item.objectID}</Text>
        <Text style={displayPageStyles.text}>Parent ID: {item.parent_id}</Text>
        <Text style={displayPageStyles.text}>Points: {item.points}</Text>
        <Text style={displayPageStyles.text}>Story ID: {item.story_id}</Text>
        <Text style={displayPageStyles.text}>
          Story Text: {item.story_text}
        </Text>
        <Text style={displayPageStyles.text}>
          Story Title: {item.story_title}
        </Text>
        <Text style={displayPageStyles.text}>Story URL: {item.story_url}</Text>
      </View>
    );
  }
}
