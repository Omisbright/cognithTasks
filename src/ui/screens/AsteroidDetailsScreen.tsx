import React, {Component} from 'react';
import {View, Text} from 'react-native';
import {AppProps} from '../../navigation/types';
import {asteroidDetailsScreenStyles} from '../styles/styles';
import axios from 'axios';

type State = {
  asteroidData: {
    name: string;
    estimated_diameter: {
      kilometers: {
        estimated_diameter_min: number;
        estimated_diameter_max: number;
      };
    };
    is_potentially_hazardous_asteroid: boolean;
  } | null;
  loading: boolean;
};

class AsteroidDetailsScreen extends Component<
  AppProps<'AsteroidDetailsScreen'>,
  State
> {
  constructor(props: AppProps<'AsteroidDetailsScreen'>) {
    super(props);
    this.state = {
      asteroidData: null,
      loading: false,
    };
  }

  componentDidMount() {
    const asteroidId = this.props.route.params?.asteroidId;
    this.fetchAsteroidDetails(asteroidId);
  }

  fetchAsteroidDetails(id: string | undefined) {
    this.setState({loading: true});
    const apiKey = 'eGo5f1A3A8Z6xlOZW1fbLBuVEO0vgJIdhWAQrjH8';
    const baseUrl = 'https://api.nasa.gov/neo/rest/v1/neo/';
    let asteroidUrl = '';

    if (id) {
      asteroidUrl = `${baseUrl}${id}?api_key=${apiKey}`;
      this.fetchAsteroidData(asteroidUrl);
    } else {
      const randomAsteroidUrl = `${baseUrl}browse?api_key=${apiKey}`;
      axios
        .get(randomAsteroidUrl)
        .then(response => {
          this.setState({loading: false});
          const data = response.data;
          const randomIndex = Math.floor(
            Math.random() * data.near_earth_objects.length,
          );
          const randomAsteroid = data.near_earth_objects[randomIndex];
          asteroidUrl = `${baseUrl}${randomAsteroid.id}?api_key=${apiKey}`;
          this.fetchAsteroidData(asteroidUrl);
        })
        .catch(error => {
          this.setState({loading: false});
          console.log('error', error);
        });
    }
  }

  fetchAsteroidData(asteroidUrl: string) {
    this.setState({loading: true});
    axios
      .get(asteroidUrl)
      .then(asteroidResponse => {
        this.setState({loading: false});
        const asteroidData = asteroidResponse.data;
        this.setState({asteroidData});
      })
      .catch(error => {
        this.setState({loading: false});
        console.log('error', error);
      });
  }

  render() {
    const {asteroidData, loading} = this.state;

    if (loading) {
      return (
        <View style={asteroidDetailsScreenStyles.container}>
          <Text>Loading...</Text>
        </View>
      );
    }

    if (!asteroidData) {
      return (
        <View style={asteroidDetailsScreenStyles.container}>
          <Text>No asteroid data found</Text>
        </View>
      );
    }

    return (
      <View style={asteroidDetailsScreenStyles.container}>
        <Text>Name: {asteroidData.name}</Text>
        <Text>
          Estimated Diameter:{' '}
          {asteroidData.estimated_diameter.kilometers.estimated_diameter_min} -{' '}
          {asteroidData.estimated_diameter.kilometers.estimated_diameter_max} km
        </Text>
        <Text>
          Potentially Hazardous:{' '}
          {asteroidData.is_potentially_hazardous_asteroid ? 'Yes' : 'No'}
        </Text>
      </View>
    );
  }
}

export default AsteroidDetailsScreen;
