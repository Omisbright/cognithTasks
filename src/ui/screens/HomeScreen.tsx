import React, {Component} from 'react';
import {Alert, Button, Text, TextInput, View} from 'react-native';
import {AppProps} from '../../navigation/types';
import {homeScreenStyles} from '../styles/styles';

type State = {
  asteroidId: string;
};
class HomeScreen extends Component<AppProps<'HomeScreen'>, State> {
  constructor(props: AppProps<'HomeScreen'>) {
    super(props);
    this.state = {
      asteroidId: '',
    };
  }

  handleSubmit = () => {
    const {asteroidId} = this.state;
    if (!asteroidId) {
      Alert.alert(
        'Error',
        'Please enter an asteroid ID',
        [{text: 'OK', onPress: () => console.log('OK pressed')}],
        {cancelable: false},
      );
    } else {
      this.props.navigation.navigate('AsteroidDetailsScreen', {
        asteroidId: this.state.asteroidId,
      });
    }
  };

  render() {
    return (
      <View style={homeScreenStyles.container}>
        <Text style={homeScreenStyles.text}>Enter Asteroid ID:</Text>
        <TextInput
          testID="textInput"
          style={homeScreenStyles.textInput}
          onChangeText={text => this.setState({asteroidId: text})}
          value={this.state.asteroidId}
        />
        <Button
          testID="submitButton"
          title="Submit"
          onPress={() => this.handleSubmit()}
        />
        <Button
          testID="randomAsteroidNavigateButton"
          title="Random Asteroid"
          onPress={() =>
            this.props.navigation.navigate('AsteroidDetailsScreen')
          }
        />
      </View>
    );
  }
}

export default HomeScreen;
