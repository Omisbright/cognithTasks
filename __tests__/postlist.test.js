import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import axios from 'axios';
import PostListComponent from '../src/screens/PostList';
import {PostList} from '../src/screens/PostList';
import {Text} from 'react-native';

// Mock the axios.get method to return mock data
jest.mock('axios');

axios.get = jest.fn(() =>
  Promise.resolve({
    data: {
      hits: [
        {objectID: '1', title: 'Post 1'},
        {objectID: '2', title: 'Post 2'},
      ],
      nbPages: 2,
    },
  }),
);

describe('PostListComponent', () => {
  const initialProps = {
    navigation: {
      resetNavigation: jest.fn(),
    },
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  // Test 1: Renders without crashing
  it('renders without crashing', () => {
    const {getByPlaceholderText} = render(
      <PostListComponent {...initialProps} />,
    );
    expect(getByPlaceholderText('Search posts...')).toBeTruthy();
  });

  // Test 2: Renders correctly
  it('renders correctly', () => {
    const tree = render(<PostListComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Test 3: Fetches posts when mounted
  it('fetches posts when mounted', async () => {
    render(<PostListComponent {...initialProps} />);
    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith(
      'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0',
    );
  });

  // Test 4: Updates displayed posts when search text changes
  it('updates displayed posts when search text changes', () => {
    const {getByPlaceholderText} = render(
      <PostListComponent {...initialProps} />,
    );
    const searchInput = getByPlaceholderText('Search posts...');
    fireEvent.changeText(searchInput, 'Example Post 3');
    expect(searchInput.props.value).toEqual('Example Post 3');
  });

  // Test 5: Updates displayed posts when page changes
  it('updates displayed posts when page changes', () => {
    const {getByText} = render(<PostListComponent {...initialProps} />);
    fireEvent.press(getByText('Next'));
    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith(
      'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0',
    );
  });

  // Test 6: Handles empty search text
  it('handles empty search text', () => {
    const {getByPlaceholderText, UNSAFE_getAllByType} = render(
      <PostList {...initialProps} />,
    );
    const searchInput = getByPlaceholderText('Search posts...');

    const initialDisplayedPosts = UNSAFE_getAllByType(Text);
    const initialPostTitles = initialDisplayedPosts.map(
      textComponent => textComponent.props.children,
    );

    fireEvent.changeText(searchInput, '');

    const updatedDisplayedPosts = UNSAFE_getAllByType(Text);
    const updatedPostTitles = updatedDisplayedPosts.map(
      textComponent => textComponent.props.children,
    );

    expect(updatedPostTitles).toEqual(initialPostTitles);
  });

  // Test 7: Handles errors when fetching posts
  it('handles errors when fetching posts', async () => {
    axios.get.mockRejectedValueOnce(new Error('Failed to fetch posts'));
    const {getByTestId} = render(<PostList />);
    await waitFor(() => getByTestId('error-message'));
    expect(getByTestId('error-message').props.children).toEqual(
      'Failed to fetch posts',
    );
  });

  // Test 8: Handles pagination correctly
  it('handles pagination correctly', () => {
    const {getByText} = render(<PostListComponent {...initialProps} />);
    getByText('Next');
    getByText('First');
    getByText('Last');
  });

  // Test 9: Fetches posts when pagination page is changed
  it('fetches posts when pagination page is changed', async () => {
    const response = {
      data: {
        hits: [
          {objectID: '1', title: 'Post 1'},
          {objectID: '2', title: 'Post 2'},
        ],
        nbPages: 3,
      },
    };
    axios.get.mockResolvedValueOnce(response);

    const {getByText} = render(<PostList {...initialProps} />);

    await waitFor(() => expect(getByText('Next')).toBeTruthy());

    fireEvent.press(getByText('Next'));
    fireEvent.press(getByText('Previous'));

    await waitFor(() => {
      expect(axios.get).toHaveBeenCalledTimes(3);
    });

    await waitFor(() => {
      expect(axios.get).toHaveBeenNthCalledWith(
        1,
        'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0',
      );
      expect(axios.get).toHaveBeenNthCalledWith(
        2,
        'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=1',
      );
      expect(axios.get).toHaveBeenNthCalledWith(
        3,
        'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0',
      );
    });
  });
});
