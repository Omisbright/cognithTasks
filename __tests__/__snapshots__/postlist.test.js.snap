// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListComponent renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 16,
    }
  }
>
  <View>
    <TextInput
      onChangeText={[Function]}
      placeholder="Search posts..."
      style={
        {
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "paddingLeft": 10,
          "paddingRight": 10,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "red",
            }
          }
        >
          List is Empty
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": 5,
        "paddingTop": 4,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Previous
      </Text>
    </View>
    <Text>
      1 of 0
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Next
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        First
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text>
        Last
      </Text>
    </View>
  </View>
</View>
`;
